src/moves.c

```diff
@@ -174,103 +174,43 @@ u8 GetMoveType(struct Move *move)
     return gMovesData[move->id].type;
 }
 
-#ifdef NONMATCHING
 u8 *GetLevelUpMoves(s16 species)
 {
-    if (species == MONSTER_DECOY || species == MONSTER_NONE)
+    s32 species_s32, species_s32_1 = species;
+#ifndef NONMATCHING    
+    asm("add \tr2, r1, #0");
+#else
+    asm("add \tr2, r1, #0");
```

Comment:
Shouldn't the `NONMATCHING` clause be empty? The ASM hack is not needed if you don't care about matching.

Reply:
https://staging.cexplore.karathan.at/z/gAOR2w

From testing, I thought it cause UB because R2 is never cleared but then compared against

Reply:
In this C code, `species_s32` is never assigned but it's later compared against, so its value is arbitrary. If you assign `species_s32` to `species` then r2 is assigned properly.

Reply:
Reverted that addition.

Reply:
Hmm, since `species_s32` remains unassigned, the `NONMATCHING` now doesn't match in functionality, and a garbage value is compared against `MONSTER_MUNCHLAX` instead of `species`. This might need something like:
```
#ifndef NONMATCHING
    s32 species_s32, species_s32_1 = species;
    asm("add \tr2, r1, #0");
#else
    s32 species_s32 = species, species_s32_1 = species;
#endif
```

Reply:
Applied.